packages/neuron-ui/src/widgets/MigrateCkbDataDialog/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { isSuccessResponse } from 'utils'
+import { Attention } from 'widgets/Icons/icon'
+import Button from 'widgets/Button'
+import AlertDialog from 'widgets/AlertDialog'
+import styles from './migrateCkbDataDialog.module.scss'
+
+const MigrateCkbDataDialog = ({
+  show,
+  prevPath,
+  currentPath,
+  onCancel,
+  onConfirm,
+}: {
+  show: boolean
+  prevPath: string
+  currentPath: string
+  onCancel: () => void
+  onConfirm: (dataPath: string) => void
+}) => {
+  const [t] = useTranslation()
+  const resyncRef = useRef<HTMLButtonElement | null>(null)
+  const [faidMessage, setFaidMessage] = useState('')
+  const [isSaving, setIsSaving] = useState(false)
+  const [savingType, setSavingType] = useState<string | null>()
+  const onStartSync = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      setFaidMessage('')
+      const { dataset } = e.currentTarget
+      setIsSaving(true)
+      setSavingType(dataset.syncType)
+      setCkbNodeDataPath({
+        dataPath: currentPath,
+        clearCache: dataset?.resync === 'true',
+      })
+        .then(res => {
+          if (isSuccessResponse(res)) {
+            onConfirm(currentPath)
+          } else {
+            setFaidMessage(typeof res.message === 'string' ? res.message : res.message.content!)
+          }
+        })
+        .finally(() => {
+          setIsSaving(false)
+          setSavingType(null)
+        })
+    },
+    [currentPath, onConfirm]
+  )
+  return (
+    <>
+      <Dialog show={show} title={t('settings.data.ckb-node-storage')} onCancel={onCancel} showFooter={false}>
+        <div className={styles.dialogContainer}>
+          <div>{t('settings.data.remove-ckb-data-tip', { prevPath, currentPath })}</div>
+          <div className={styles.attention}>
+            <Attention />
+            {t('settings.data.resync-ckb-node-describe')}
+          </div>
+
+          <div className={styles.footer}>
+            <Button
+              ref={resyncRef}
+              disabled={isSaving}
+              className={styles.footerBtn}
+              data-sync-type="resync"
+              data-resync="true"
+              label={t('settings.data.re-sync')}
+              type="primary"
+              onClick={onStartSync}
+              loading={isSaving && savingType === 'resync'}
+            />
+
+            <Button
+              disabled={isSaving}
+              className={styles.footerBtn}
+              data-sync-type="move"
+              label={t('settings.data.move-data-finish')}
+              type="primary"
+              onClick={onStartSync}
+              loading={isSaving && savingType === 'move'}
+            />
+          </div>
+        </div>
+      </Dialog>
+      <AlertDialog
+        show={!!faidMessage}
+        title={t('settings.data.ckb-node-data')}
+        message={faidMessage}
+        type="warning"
+        onCancel={() => setFaidMessage('')}
+        onOk={() => {
+          resyncRef?.current?.click()
```

Comment:
Because the `MigrateDataDialog` will be used in two components, I moved it from `packages/neuron-ui/src/components/DataSetting/index.tsx` to widgets. I will fix it.